code_generator/GdkCodeGenerator/Templates/UnityGameObjectComponentDispatcherGenerator.tt

```diff
@@ -55,26 +57,57 @@ namespace <#= qualifiedNamespace #>
 <# } #>
             };
 
-            protected override uint GetComponentId()
+            private const uint componentId = <#= unityComponentDefinition.Id #>;
+
+            public override void MarkComponentsAddedForActivation(Dictionary<int, MonoBehaviourActivationManager> entityIndexToManagers)
+            {
+                var entities = ComponentAddedComponentGroup.GetEntityArray();
```

Comment:
Check if the group is empty first. `ComponentAddedComponentGroup.IsEmptyIgnoreFilter` if true, just early out.